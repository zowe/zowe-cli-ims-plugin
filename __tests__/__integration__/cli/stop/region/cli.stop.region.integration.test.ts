/*
* This program and the accompanying materials are made available under the terms of the *
* Eclipse Public License v2.0 which accompanies this distribution, and is available at *
* https://www.eclipse.org/legal/epl-v20.html                                      *
*                                                                                 *
* SPDX-License-Identifier: EPL-2.0                                                *
*                                                                                 *
* Copyright Contributors to the Zowe Project.                                     *
*                                                                                 *
*/

import { TestEnvironment } from "../../../../__src__/environment/TestEnvironment";
import { ITestEnvironment } from "../../../../__src__/environment/doc/response/ITestEnvironment";
import { runCliScript } from "../../../../__src__/TestUtils";
import { join } from "path";

let testEnvironment: ITestEnvironment;
describe("Start region command", () => {

    beforeAll(async () => {
        testEnvironment = await TestEnvironment.setUp({
            testName: "start_region_cli_integration",
            installPlugin: true,
            skipProperties: true
        });
    });

    afterAll(async () => {
        await TestEnvironment.cleanUp(testEnvironment);
    });

    it("should display the help", async () => {
        const result = runCliScript(join(__dirname, "__scripts__", "stop_region_help.sh"), testEnvironment);
        expect(result.stderr.toString()).toEqual("");
        expect(result.status).toEqual(0);
        expect(result.stdout.toString()).toMatchSnapshot();
    });

    it("should reject mutually exclusive options --job-name and --region-ids", async () => {
        const result = runCliScript(join(__dirname, "__scripts__", "stop_region_conflict.sh"), testEnvironment);
        expect(result.status).toEqual(1);
        expect(result.stderr.toString()).toContain("conflict");
        expect(result.stderr.toString()).toMatchSnapshot();
    });

});
