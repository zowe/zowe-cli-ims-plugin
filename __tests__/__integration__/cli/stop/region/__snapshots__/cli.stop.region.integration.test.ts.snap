// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Start region command should display the help 1`] = `
"
 COMMAND NAME
 ------------

   region | reg

 DESCRIPTION
 -----------

   Stop an IMS region.

 USAGE
 -----

   zowe ims stop region [options]

 OPTIONS
 -------

   --region-ids  | --ri (array)

      Region identifier numbers for the regions you want to stop. You must specify
      either this option or --job-name.

   --job-name  | --jn (string)

      The name of the job for the IMS region you want to stop. You must specify either
      this option or --region-ids.

   --route  | --rte (array)

      The region(s) to route the command to

   --abdump  (string)

      Specify this option to cause abnormal termination (ABEND) of an application
      program. If the transaction indicated by this argument is currently running in
      the specified region, an error message is received at the master terminal,
      indicating an application program ABEND. The region will remain active, but the
      transaction will be stopped. The command is ignored if the transaction is not
      currently scheduled in the region.

   --cancel  (boolean)

      Use this option if the region cannot be stopped with a stop region --abdump
      command. To use this option, you must have already submitted a stop region
      command using the --abdump option.

   --transaction  (string)

      Specify a transaction in wait-for-input mode to stop its message processing
      within the specified region.

 IMS CONNECTION OPTIONS
 ----------------------

   --host  | -H (string)

      The IMS Operations API server host name.

   --port  | -P (number)

      The IMS Operations API server port.

   --ims-connect-host  | --ich (string)

      The hostname of your instance of IMS Connect. This is typically the hostname of
      the mainframe LPAR where IMS Connect is running.

   --ims-connect-port  | --icp (number)

      The port of your instance of IMS Connect. This port can be found in your IMS
      Connect configuration file on the mainframe.

   --plex  | -x (string)

      The name of the IMS plex.

   --user  | -u (string)

      The web server user name where the IMS Operations API resides.

   --password  | --pass (string)

      The web server user password where the IMS Operations API resides.

   --base-path  | --bp (string)

      The base path for your API mediation layer instance. Specify this option to
      prepend the base path to all z/OSMF resources when making REST requests. Do not
      specify this option if you are not using an API mediation layer.

 PROFILE OPTIONS
 ---------------

   --ims-profile  | --ims-p (string)

      The name of a (ims) profile to load for this command execution.

   --base-profile  | --base-p (string)

      The name of a (base) profile to load for this command execution.

 BASE CONNECTION OPTIONS
 -----------------------

   --reject-unauthorized  | --ru (boolean)

      Reject self-signed certificates.

      Default value: true

   --token-type  | --tt (string)

      The type of token to get and use for the API. Omit this option to use the
      default token type, which is provided by 'zowe auth login'.

   --token-value  | --tv (string)

      The value of the token to pass to the API.

 RESPONSE FORMAT OPTIONS
 -----------------------

   --response-format-filter  | --rff (array)

      Filter (include) fields in the response. Accepts an array of field/property
      names to include in the output response. You can filter JSON objects properties
      OR table columns/fields. In addition, you can use this option in conjunction
      with '--response-format-type' to reduce the output of a command to a single
      field/property or a list of a single field/property.

   --response-format-type  | --rft (string)

      The command response output format type. Must be one of the following:

      table: Formats output data as a table. Use this option when the output data is
      an array of homogeneous JSON objects. Each property of the object will become a
      column in the table.

      list: Formats output data as a list of strings. Can be used on any data type
      (JSON objects/arrays) are stringified and a new line is added after each entry
      in an array.

      object: Formats output data as a list of prettified objects (or single object).
      Can be used in place of \\"table\\" to change from tabular output to a list of
      prettified objects.

      string: Formats output data as a string. JSON objects/arrays are stringified.

      Allowed values: table, list, object, string

   --response-format-header  | --rfh (boolean)

      If \\"--response-format-type table\\" is specified, include the column headers in
      the output.

 GLOBAL OPTIONS
 --------------

   --response-format-json  | --rfj (boolean)

      Produce JSON formatted data from a command

   --help  | -h (boolean)

      Display help text

   --help-examples  (boolean)

      Display examples for all the commands in a the group

   --help-web  | --hw (boolean)

      Display HTML help in browser

 EXAMPLES
 --------

   - Stop a region with job name JOBNM1:

      $ zowe ims stop region --job-name \\"JOBNM1\\"

   - Stop multiple regions with region identifiers:

      $ zowe ims stop region --region-ids 4 5

   - Stop a region with region identifier and cause the abnormal
   termination (ABEND) of the application program:

      $ zowe ims stop region --region-ids 4 --abdump \\"TRAN1\\"

   - Stop a region with region identifier and specify 'cancel'
   because the 'abdump' option failed to stop the region:

      $ zowe ims stop region --region-ids 4 --cancel true

   - Stop a region with job name JOBNM4 specifying optional
   connection parameters:

      $ zowe ims stop region --job-name \\"JOBNM4\\" --user \\"username\\" --pass \\"pass1234\\" --host \\"localhost\\" --port 8080--ich \\"zos1\\" --icp 9999 --plex \\"PLEX1\\"

{
  \\"success\\": true,
  \\"exitCode\\": 0,
  \\"message\\": \\"The help was constructed for command: region.\\",
  \\"stdout\\": \\"\\\\n COMMAND NAME\\\\n ------------\\\\n\\\\n   region | reg\\\\n\\\\n DESCRIPTION\\\\n -----------\\\\n\\\\n   Stop an IMS region.\\\\n\\\\n USAGE\\\\n -----\\\\n\\\\n   zowe ims stop region [options]\\\\n\\\\n OPTIONS\\\\n -------\\\\n\\\\n   --region-ids  | --ri (array)\\\\n\\\\n      Region identifier numbers for the regions you want to stop. You must specify\\\\n      either this option or --job-name.\\\\n\\\\n   --job-name  | --jn (string)\\\\n\\\\n      The name of the job for the IMS region you want to stop. You must specify either\\\\n      this option or --region-ids.\\\\n\\\\n   --route  | --rte (array)\\\\n\\\\n      The region(s) to route the command to\\\\n\\\\n   --abdump  (string)\\\\n\\\\n      Specify this option to cause abnormal termination (ABEND) of an application\\\\n      program. If the transaction indicated by this argument is currently running in\\\\n      the specified region, an error message is received at the master terminal,\\\\n      indicating an application program ABEND. The region will remain active, but the\\\\n      transaction will be stopped. The command is ignored if the transaction is not\\\\n      currently scheduled in the region.\\\\n\\\\n   --cancel  (boolean)\\\\n\\\\n      Use this option if the region cannot be stopped with a stop region --abdump\\\\n      command. To use this option, you must have already submitted a stop region\\\\n      command using the --abdump option.\\\\n\\\\n   --transaction  (string)\\\\n\\\\n      Specify a transaction in wait-for-input mode to stop its message processing\\\\n      within the specified region.\\\\n\\\\n IMS CONNECTION OPTIONS\\\\n ----------------------\\\\n\\\\n   --host  | -H (string)\\\\n\\\\n      The IMS Operations API server host name.\\\\n\\\\n   --port  | -P (number)\\\\n\\\\n      The IMS Operations API server port.\\\\n\\\\n   --ims-connect-host  | --ich (string)\\\\n\\\\n      The hostname of your instance of IMS Connect. This is typically the hostname of\\\\n      the mainframe LPAR where IMS Connect is running.\\\\n\\\\n   --ims-connect-port  | --icp (number)\\\\n\\\\n      The port of your instance of IMS Connect. This port can be found in your IMS\\\\n      Connect configuration file on the mainframe.\\\\n\\\\n   --plex  | -x (string)\\\\n\\\\n      The name of the IMS plex.\\\\n\\\\n   --user  | -u (string)\\\\n\\\\n      The web server user name where the IMS Operations API resides.\\\\n\\\\n   --password  | --pass (string)\\\\n\\\\n      The web server user password where the IMS Operations API resides.\\\\n\\\\n   --base-path  | --bp (string)\\\\n\\\\n      The base path for your API mediation layer instance. Specify this option to\\\\n      prepend the base path to all z/OSMF resources when making REST requests. Do not\\\\n      specify this option if you are not using an API mediation layer.\\\\n\\\\n PROFILE OPTIONS\\\\n ---------------\\\\n\\\\n   --ims-profile  | --ims-p (string)\\\\n\\\\n      The name of a (ims) profile to load for this command execution.\\\\n\\\\n   --base-profile  | --base-p (string)\\\\n\\\\n      The name of a (base) profile to load for this command execution.\\\\n\\\\n BASE CONNECTION OPTIONS\\\\n -----------------------\\\\n\\\\n   --reject-unauthorized  | --ru (boolean)\\\\n\\\\n      Reject self-signed certificates.\\\\n\\\\n      Default value: true\\\\n\\\\n   --token-type  | --tt (string)\\\\n\\\\n      The type of token to get and use for the API. Omit this option to use the\\\\n      default token type, which is provided by 'zowe auth login'.\\\\n\\\\n   --token-value  | --tv (string)\\\\n\\\\n      The value of the token to pass to the API.\\\\n\\\\n RESPONSE FORMAT OPTIONS\\\\n -----------------------\\\\n\\\\n   --response-format-filter  | --rff (array)\\\\n\\\\n      Filter (include) fields in the response. Accepts an array of field/property\\\\n      names to include in the output response. You can filter JSON objects properties\\\\n      OR table columns/fields. In addition, you can use this option in conjunction\\\\n      with '--response-format-type' to reduce the output of a command to a single\\\\n      field/property or a list of a single field/property.\\\\n\\\\n   --response-format-type  | --rft (string)\\\\n\\\\n      The command response output format type. Must be one of the following:\\\\n\\\\n      table: Formats output data as a table. Use this option when the output data is\\\\n      an array of homogeneous JSON objects. Each property of the object will become a\\\\n      column in the table.\\\\n\\\\n      list: Formats output data as a list of strings. Can be used on any data type\\\\n      (JSON objects/arrays) are stringified and a new line is added after each entry\\\\n      in an array.\\\\n\\\\n      object: Formats output data as a list of prettified objects (or single object).\\\\n      Can be used in place of \\\\\\"table\\\\\\" to change from tabular output to a list of\\\\n      prettified objects.\\\\n\\\\n      string: Formats output data as a string. JSON objects/arrays are stringified.\\\\n\\\\n      Allowed values: table, list, object, string\\\\n\\\\n   --response-format-header  | --rfh (boolean)\\\\n\\\\n      If \\\\\\"--response-format-type table\\\\\\" is specified, include the column headers in\\\\n      the output.\\\\n\\\\n GLOBAL OPTIONS\\\\n --------------\\\\n\\\\n   --response-format-json  | --rfj (boolean)\\\\n\\\\n      Produce JSON formatted data from a command\\\\n\\\\n   --help  | -h (boolean)\\\\n\\\\n      Display help text\\\\n\\\\n   --help-examples  (boolean)\\\\n\\\\n      Display examples for all the commands in a the group\\\\n\\\\n   --help-web  | --hw (boolean)\\\\n\\\\n      Display HTML help in browser\\\\n\\\\n EXAMPLES\\\\n --------\\\\n\\\\n   - Stop a region with job name JOBNM1:\\\\n\\\\n      $ zowe ims stop region --job-name \\\\\\"JOBNM1\\\\\\"\\\\n\\\\n   - Stop multiple regions with region identifiers:\\\\n\\\\n      $ zowe ims stop region --region-ids 4 5\\\\n\\\\n   - Stop a region with region identifier and cause the abnormal\\\\n   termination (ABEND) of the application program:\\\\n\\\\n      $ zowe ims stop region --region-ids 4 --abdump \\\\\\"TRAN1\\\\\\"\\\\n\\\\n   - Stop a region with region identifier and specify 'cancel'\\\\n   because the 'abdump' option failed to stop the region:\\\\n\\\\n      $ zowe ims stop region --region-ids 4 --cancel true\\\\n\\\\n   - Stop a region with job name JOBNM4 specifying optional\\\\n   connection parameters:\\\\n\\\\n      $ zowe ims stop region --job-name \\\\\\"JOBNM4\\\\\\" --user \\\\\\"username\\\\\\" --pass \\\\\\"pass1234\\\\\\" --host \\\\\\"localhost\\\\\\" --port 8080--ich \\\\\\"zos1\\\\\\" --icp 9999 --plex \\\\\\"PLEX1\\\\\\"\\\\n\\\\n\\",
  \\"stderr\\": \\"\\",
  \\"data\\": \\"\\\\n COMMAND NAME\\\\n ------------\\\\n\\\\n   region | reg\\\\n\\\\n DESCRIPTION\\\\n -----------\\\\n\\\\n   Stop an IMS region.\\\\n\\\\n USAGE\\\\n -----\\\\n\\\\n   zowe ims stop region [options]\\\\n\\\\n OPTIONS\\\\n -------\\\\n\\\\n   --region-ids  | --ri (array)\\\\n\\\\n      Region identifier numbers for the regions you want to stop. You must specify\\\\n      either this option or --job-name.\\\\n\\\\n   --job-name  | --jn (string)\\\\n\\\\n      The name of the job for the IMS region you want to stop. You must specify either\\\\n      this option or --region-ids.\\\\n\\\\n   --route  | --rte (array)\\\\n\\\\n      The region(s) to route the command to\\\\n\\\\n   --abdump  (string)\\\\n\\\\n      Specify this option to cause abnormal termination (ABEND) of an application\\\\n      program. If the transaction indicated by this argument is currently running in\\\\n      the specified region, an error message is received at the master terminal,\\\\n      indicating an application program ABEND. The region will remain active, but the\\\\n      transaction will be stopped. The command is ignored if the transaction is not\\\\n      currently scheduled in the region.\\\\n\\\\n   --cancel  (boolean)\\\\n\\\\n      Use this option if the region cannot be stopped with a stop region --abdump\\\\n      command. To use this option, you must have already submitted a stop region\\\\n      command using the --abdump option.\\\\n\\\\n   --transaction  (string)\\\\n\\\\n      Specify a transaction in wait-for-input mode to stop its message processing\\\\n      within the specified region.\\\\n\\\\n IMS CONNECTION OPTIONS\\\\n ----------------------\\\\n\\\\n   --host  | -H (string)\\\\n\\\\n      The IMS Operations API server host name.\\\\n\\\\n   --port  | -P (number)\\\\n\\\\n      The IMS Operations API server port.\\\\n\\\\n   --ims-connect-host  | --ich (string)\\\\n\\\\n      The hostname of your instance of IMS Connect. This is typically the hostname of\\\\n      the mainframe LPAR where IMS Connect is running.\\\\n\\\\n   --ims-connect-port  | --icp (number)\\\\n\\\\n      The port of your instance of IMS Connect. This port can be found in your IMS\\\\n      Connect configuration file on the mainframe.\\\\n\\\\n   --plex  | -x (string)\\\\n\\\\n      The name of the IMS plex.\\\\n\\\\n   --user  | -u (string)\\\\n\\\\n      The web server user name where the IMS Operations API resides.\\\\n\\\\n   --password  | --pass (string)\\\\n\\\\n      The web server user password where the IMS Operations API resides.\\\\n\\\\n   --base-path  | --bp (string)\\\\n\\\\n      The base path for your API mediation layer instance. Specify this option to\\\\n      prepend the base path to all z/OSMF resources when making REST requests. Do not\\\\n      specify this option if you are not using an API mediation layer.\\\\n\\\\n PROFILE OPTIONS\\\\n ---------------\\\\n\\\\n   --ims-profile  | --ims-p (string)\\\\n\\\\n      The name of a (ims) profile to load for this command execution.\\\\n\\\\n   --base-profile  | --base-p (string)\\\\n\\\\n      The name of a (base) profile to load for this command execution.\\\\n\\\\n BASE CONNECTION OPTIONS\\\\n -----------------------\\\\n\\\\n   --reject-unauthorized  | --ru (boolean)\\\\n\\\\n      Reject self-signed certificates.\\\\n\\\\n      Default value: true\\\\n\\\\n   --token-type  | --tt (string)\\\\n\\\\n      The type of token to get and use for the API. Omit this option to use the\\\\n      default token type, which is provided by 'zowe auth login'.\\\\n\\\\n   --token-value  | --tv (string)\\\\n\\\\n      The value of the token to pass to the API.\\\\n\\\\n RESPONSE FORMAT OPTIONS\\\\n -----------------------\\\\n\\\\n   --response-format-filter  | --rff (array)\\\\n\\\\n      Filter (include) fields in the response. Accepts an array of field/property\\\\n      names to include in the output response. You can filter JSON objects properties\\\\n      OR table columns/fields. In addition, you can use this option in conjunction\\\\n      with '--response-format-type' to reduce the output of a command to a single\\\\n      field/property or a list of a single field/property.\\\\n\\\\n   --response-format-type  | --rft (string)\\\\n\\\\n      The command response output format type. Must be one of the following:\\\\n\\\\n      table: Formats output data as a table. Use this option when the output data is\\\\n      an array of homogeneous JSON objects. Each property of the object will become a\\\\n      column in the table.\\\\n\\\\n      list: Formats output data as a list of strings. Can be used on any data type\\\\n      (JSON objects/arrays) are stringified and a new line is added after each entry\\\\n      in an array.\\\\n\\\\n      object: Formats output data as a list of prettified objects (or single object).\\\\n      Can be used in place of \\\\\\"table\\\\\\" to change from tabular output to a list of\\\\n      prettified objects.\\\\n\\\\n      string: Formats output data as a string. JSON objects/arrays are stringified.\\\\n\\\\n      Allowed values: table, list, object, string\\\\n\\\\n   --response-format-header  | --rfh (boolean)\\\\n\\\\n      If \\\\\\"--response-format-type table\\\\\\" is specified, include the column headers in\\\\n      the output.\\\\n\\\\n GLOBAL OPTIONS\\\\n --------------\\\\n\\\\n   --response-format-json  | --rfj (boolean)\\\\n\\\\n      Produce JSON formatted data from a command\\\\n\\\\n   --help  | -h (boolean)\\\\n\\\\n      Display help text\\\\n\\\\n   --help-examples  (boolean)\\\\n\\\\n      Display examples for all the commands in a the group\\\\n\\\\n   --help-web  | --hw (boolean)\\\\n\\\\n      Display HTML help in browser\\\\n\\\\n EXAMPLES\\\\n --------\\\\n\\\\n   - Stop a region with job name JOBNM1:\\\\n\\\\n      $ zowe ims stop region --job-name \\\\\\"JOBNM1\\\\\\"\\\\n\\\\n   - Stop multiple regions with region identifiers:\\\\n\\\\n      $ zowe ims stop region --region-ids 4 5\\\\n\\\\n   - Stop a region with region identifier and cause the abnormal\\\\n   termination (ABEND) of the application program:\\\\n\\\\n      $ zowe ims stop region --region-ids 4 --abdump \\\\\\"TRAN1\\\\\\"\\\\n\\\\n   - Stop a region with region identifier and specify 'cancel'\\\\n   because the 'abdump' option failed to stop the region:\\\\n\\\\n      $ zowe ims stop region --region-ids 4 --cancel true\\\\n\\\\n   - Stop a region with job name JOBNM4 specifying optional\\\\n   connection parameters:\\\\n\\\\n      $ zowe ims stop region --job-name \\\\\\"JOBNM4\\\\\\" --user \\\\\\"username\\\\\\" --pass \\\\\\"pass1234\\\\\\" --host \\\\\\"localhost\\\\\\" --port 8080--ich \\\\\\"zos1\\\\\\" --icp 9999 --plex \\\\\\"PLEX1\\\\\\"\\\\n\\\\n\\"
}"
`;

exports[`Start region command should reject mutually exclusive options --job-name and --region-ids 1`] = `
"
Syntax Error:
The following options conflict (mutually exclusive):
--region-ids
--job-name

Syntax Error:
The following options conflict (mutually exclusive):
--job-name
--region-ids

Syntax Error:
Missing Required Option:
--host (-H)

Option Description:
The IMS Operations API server host name.

Syntax Error:
Missing Required Option:
--port (-P)

Option Description:
The IMS Operations API server port.

Syntax Error:
Missing Required Option:
--ims-connect-host (--ich)

Option Description:
The hostname of your instance of IMS Connect. This is typically the hostname of
the mainframe LPAR where IMS Connect is running.

Syntax Error:
Missing Required Option:
--ims-connect-port (--icp)

Option Description:
The port of your instance of IMS Connect. This port can be found in your IMS
Connect configuration file on the mainframe.

Syntax Error:
Missing Required Option:
--plex (-x)

Option Description:
The name of the IMS plex.

Example:

 - Stop a region with job name JOBNM1:

      $ zowe ims stop region --job-name \\"JOBNM1\\"

Use \\"zowe ims stop region --help\\" to view command description, usage, and options.
"
`;
